### PR TITLE
Bluetooth: Host: Add note in config BT_PRIVACY help for dir-adv

### PR DESCRIPTION
**Pull request by alwa-nordic**
See https://github.com/zephyrproject-rtos/zephyr/pull/53961#pullrequestreview-1267549957

Signed-off-by: Aleksander Wasaznik <aleksander.wasaznik@nordicsemi.no>
Co-authored-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/host/Kconfig [hunk 1]
```diff
@@ -330,6 +330,10 @@ config BT_PRIVACY
 	  Enable local Privacy Feature support. This makes it possible
 	  to use Resolvable Private Addresses (RPAs).
 
+	  Note: Without privacy support in the controller establishing
+	  connections as a directed advertiser, or to a directed advertiser is
+	  not possible.
+
```
Comment (Thalley) [inline subsys/bluetooth/host/Kconfig:336]: What about the comment https://github.com/zephyrproject-rtos/zephyr/pull/53961#issuecomment-1403583426? 

Is it clear that `BT_PRIVACY` is host-based privacy in this Kconfig?

Comment (alwa-nordic) [reply subsys/bluetooth/host/Kconfig:336]: Right. The comment is unresolved. I just though we should move the discussion to a new PR.

Comment (alwa-nordic) [reply subsys/bluetooth/host/Kconfig:336]: I think the most visible place for the note is in the connect/advertise API doc. Visible is good, I think.

Comment (Thalley) [reply subsys/bluetooth/host/Kconfig:336]: > I think the most visible place for the note is in the connect/advertise API doc. Visible is good, I think.

I agree. If we move it to the API, then it is generally more visible, and we can use `@note` or `@warning` to make it very clear, and we can put it at `BT_LE_ADV_OPT_DIR_ADDR_RPA` for the advertiser

Comment (joerchan) [reply subsys/bluetooth/host/Kconfig:336]: I still think it should be documented like this:

```
config BT_PRIVACY
	bool "Enable Privacy"
	help
	  Enable Privacy for the local device. This makes the local device
	  use Resolvable Private Addresses (RPAs).
```
Maybe it wasn't clear what I meant, but I suggested updating the prompt, and the help text.
It should better describe that it is about ENABLING privacy, not enabling support for privacy in the host.
When you enable this option, you will be using RPAs.

As for the note, that is important information. But it is not clear to me where we should be documenting it.
Maybe just have it on BT_CTLR_PRIVACY, now that I had some time to think on it.
You need it to start a directed advertiser to a privacy enabled peer, even if you are not yourself privacy-enabled.

**Comment by jori-nordic** [comment]
@alwa-nordic ping

**Review by jori-nordic — APPROVED** [review]

File: subsys/bluetooth/host/Kconfig (hunk 1 above)
Comment (jori-nordic) [reply subsys/bluetooth/host/Kconfig:336]: @joerchan is the wording in #58120 correct for you?

**Comment by alwa-nordic** [comment]
Superseded by https://github.com/zephyrproject-rtos/zephyr/pull/58120

**Pull request closed without merging**